k3s-io/k3s
# Check for `--kubeconfig` flag with embedded `kubectl`

#### Proposed Changes ####
Seeks to address parts of the issue with #1541, by checking for a `--kubeconfig` flag argument. This prevent incorrect [Warning] logs from happening. 


#### Types of Changes ####
Bugfix


#### Verification ####
- Start up k3s
- Copy out the `/etc/rancher/k3s/k3s.yaml` into another location ( `/tmp/tempconfig` in this example)
- Check that `k3s kubectl get pods -A --kubeconfig /tmp//tempconfig` executes without warnings
- Check that `k3s kubectl get pods -A --kubeconfig=/tmp/tempconfig` executes without warnings


#### Linked Issues ####
https://github.com/k3s-io/k3s/issues/1541


#### User-Facing Change ####

```release-note
K3s kubectl now no longer outputs a [Warning] when using the `--kubeconfig` flag with a non-default config location
```

#### Further Comments ####
